cmd/frontend/internal/app/jscontext/jscontext.go

```diff
@@ -378,6 +403,38 @@ func derefString(s *string) string {
 	return *s
 }
 
+func resolverUserPermissions(ctx context.Context, userResolver *graphqlbackend.UserResolver, licenseInfo *hooks.LicenseInfo) *PermissionsConnection {
+	if isFreePlan(licenseInfo) {
+		return nil
+	}
+
+	userID := userResolver.ID()
+
+	permissionResolver, err := userResolver.Permissions(ctx, &graphqlbackend.ListPermissionArgs{
+		ConnectionResolverArgs: graphqlutil.ConnectionResolverArgs{},
+		User:                   &userID,
+	})
+
+	userPermissions := []Permission{}
```

Comment:
we already know the size, we can create a slice with capacity right away:
```suggestion
	userPermissions := make([]Permission, 0, len(nodes))
```

Reply:
If there's an error, `nodes` could be nil.

Reply:
it's okay, it will be resolved to 0

Reply:
Oh nice. I didn't know that.
TIL